jni/include/faiss_stream_support.h

```diff
@@ -56,6 +56,40 @@ class FaissOpenSearchIOReader final : public faiss::IOReader {
 };  // class FaissOpenSearchIOReader
 
 
+/**
+ * A glue component inheriting IOWriter to delegate IO processing down to the given
+ * mediator. The mediator is expected to do write bytes via the provided Lucene's IndexOutput.
+ */
+class FaissOpenSearchIOWriter final : public faiss::IOWriter {
+ public:
+  explicit FaissOpenSearchIOWriter(NativeEngineIndexOutputMediator *_mediator)
+      : faiss::IOWriter(),
+        mediator(_mediator) {
```

Comment:
Can we do null check on mediator before proceeding.
```
if (!mediator) {
    throw std::invalid_argument("Mediator cannot be nullptr");
  }
```

Reply:
Mediator will be passed from wrapper, and it is not null.
but yeah, will update in the next rev